src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/StateHandler.java

```
@@ -0,0 +1,207 @@
+package com.ericsson.gerrit.plugins.eiffel.handlers;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.handlers.DataBaseHandler.Table;
+
+public class StateHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StateHandler.class);
+    private static final String DISABLED_RESPONSE = "No plugin dir given. Saving last event id is disabled for tests.";
+    private static final String FILE_ENDING = "db";
+    private final File pluginDir;
+
+    /**
+     * Constructor for the StateHandler class. in case the stateHandler is initiated
+     * with empty string or null the StateHandeling will be disabled.
+     *
+     * @param pluginDir
+     */
+    public StateHandler(final File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    /**
+     * This function returns the last set scm change eiffel event id stored based on
+     * the branch and specific project.
+     *
+     * @param project
+     * @param branch
+     * @return
+     */
+    public String getLastSourceChangeSubmittedEiffelEvent(String project, String branch) {
+        if (isDisabled()) {
+            return "";
+        }
+
+        File parentDir = new File(buildParentFilePath(project));
+        if(!(parentDir.exists())) {
+            return "";
+        }
+
+        try {
+            String fileName = String.format("%s.%s", project, FILE_ENDING);
+            DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+            String eventId = dBHandler.getEventID(Table.SCS_TABLE, branch);
+            LOGGER.info("Fetched old event with id '{}', for project '{}', and branch '{}'", eventId, project, branch);
+            return eventId;
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to get eiffel event id from database: {}\n{}", e.getMessage(), e);
+            return "";
```

Same here, i don't think we should return empty if we didnt find the event. Better to throw an error and then we can handle it however we want outside.

---

done